### ISSUE TITLE
KSP option me.tatarka.inject.generateCompanionExtensions is not working

### ISSUE THREAD

**Issue by simonlebras** [amzn/kotlin-inject-anvil#90]
When using `me.tatarka.inject.generateCompanionExtensions = true`, code generated with kotlin-inject-anvil is not compiling.

Setting the option to `true` disables the creation of 
`public fun KClass<KotlinInjectAppComponent>.create(): KotlinInjectAppComponent = InjectKotlinInjectAppComponent()`

but kotlin-inject-anvil generates a function that relies on it 
`public fun KClass<AppComponent>.create(): AppComponent = KotlinInjectAppComponent::class.create()`

**Comment by simonlebras** [comment]
Yes I do. 
kotlin-inject is generating
` public fun KotlinInjectAppComponent.Companion.create():
    KotlinInjectAppComponent = InjectKotlinInjectAppComponent()`

but kotlin-inject-anvil is generating
`public fun KClass<AppComponent>.create(): AppComponent =
    KotlinInjectAppComponent::class.create()`

which should be replaced with
`public fun AppComponent.Companion.create(): AppComponent =
    KotlinInjectAppComponent.create()`

I thought #81 would be enough but it needs more work. I already have a fix. I'll create a PR by the end of the week once I've added corresponding unit tests

**Comment by vRallev** [comment]
Is this still an issue?

**Comment by AlexBurdu** [comment]
Yep, still an issue. I ran into it this week and had to remote the kotlin-inject generateCompanionExtensions line.

**Comment by Hospes** [comment]
Same happen for me, if `me.tatarka.inject.generateCompanionExtensions` enabled compilation failed.
I'm using `0.1.6`
